### PR TITLE
Feature/spring security

### PR DESCRIPTION
**Pull request by torsello**
## Summary
- new Field on User (Password)
- Custom authentication provider
- Password encoder

## Feature screenshot
- https://i.imgur.com/mUaCCfQ.png
- https://i.imgur.com/2J7xdE2.png

## Trello Card
- https://trello.com/c/M6st9czC

### CODE REVIEW

File: src/main/java/com/wolox/training/services/UserService.java [hunk 1]
```diff
@@ -129,5 +133,6 @@ private void adaptUserDTOToUserModel(UserDTO userDTO, User user) {
         user.setUsername(userDTO.getUsername());
         user.setBirthdate(userDTO.getBirthdate());
         user.setName(userDTO.getName());
+        user.setPassword(passwordEncoder.encode(userDTO.getPassword()));
```
Comment (zeminlu) [inline src/main/java/com/wolox/training/services/UserService.java]: The password should be set in a sepparate endpoint dedicated to it. 
Remove this line, add another endpoint for doing this and adapt as necessary.

**Review by zeminlu — CHANGES_REQUESTED** [review]
Looking good!! Just one comment

File: src/main/java/com/wolox/training/config/SecurityConfig.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+import org.springframework.security.config.annotation.authentication.builders.AuthenticationManagerBuilder;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+
+@Configuration
+@EnableWebSecurity
+@ComponentScan("com.wolox.training.security")
```
Comment (zeminlu) [inline src/main/java/com/wolox/training/config/SecurityConfig.java]: I think you can remove this line.

File: src/main/java/com/wolox/training/services/UserService.java [hunk 3]
```diff
@@ -125,9 +130,27 @@ public User removeBookToAUser(Long userId, Long bookId) {
...
+     * This method sets {@link User} password
+     * @param userId: user's id
+     * @param password: new user's password
+     * @return the {@link User} updated
+     */
+    @Transactional
+    public User updateUserPassword(Long userId, String password){
+        User user = this.findOne(userId);
+        setEncodedPassword(user, password);
+        return user;
```
Comment (julianvasquez-wolox) [inline src/main/java/com/wolox/training/services/UserService.java]: I think maybe you missed the userRepository.save(user) line 😅

**Review by julianvasquez-wolox — APPROVED** [review]
👌🏼😁

**Review by zeminlu — APPROVED** [review]

**Pull request merged**